R/zchunk_L124.nonco2_unmgd_R_S_T_Y.R

```diff
@@ -43,80 +41,182 @@ module_emissions_L124.nonco2_unmgd_R_S_T_Y_DISABLED <- function(command, ...) {
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Reformat temporary data -- Note: this can be removed after the upstream files are done
+    L124.LC_bm2_R_Grass_Yh_GLU_adj %>%
+      gather(year, value, -GCAM_region_ID, -Land_Type, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L124.LC_bm2_R_Grass_Yh_GLU_adj
+
+    L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj %>%
+      gather(year, value, -GCAM_region_ID, -Land_Type, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj
+
+    # Prepare EDGAR emissions for use
+    # Map in region ID and sector name; extrapolate to 2010; aggregate to sector & region; convert from Gg to Tg
+    EDGAR_gases %>%
+      left_join(EDGAR_sector, by = "IPCC") %>%                                          # Add GCAM sector from the sector mapping
+      rename(sector = agg_sector) %>%
+      mutate(iso = tolower( ISO_A3 ), ISO_A3 = NULL) %>%
+      change_iso_code('rou', 'rom') %>%                                                 # Switch Romania iso code to its pre-2002 value
+      left_join(iso_GCAM_regID, by = "iso") %>%                                         # Map in GCAM regions
+      select(IPCC, Non.CO2, sector, iso, GCAM_region_ID, year, value) ->
+      EDGAR_history
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+      # The old data system processed data in wide format. There was a na.omit() in the code that effectively removed
+      # any region/sector combination where the time series was incomplete. We'd like to keep this information.
+      # Additionally, the old data system never included NH3 in the processing. The emissions are small
+      # but we should keep them.
+      EDGAR_history %>%
+        filter(Non.CO2 != "NH3" ) %>%                                                    # Remove NH3 for consistency with old data
+        filter(year <= 2008) %>%                                                         # Old data didn't care if post-2008 data was missing so remove it here
+        spread(year, value) %>%                                                          # Convert to wide format
+        na.omit() %>%                                                                    # Remove any row with an NA (i.e., incomplete time series)
```

Comment:
For me, this depends on how big the data are (and so how much time it takes). Kate's spread-omit-gather lines are dead simple to read and understand, so I'd only change to the filter-semi_join approach if it's a significant speed gain.

I agree this is a good candidate for refactoring, but that could happen later. Kate, whether you do this or not, could you open an issue on this, citing this code chunk?